src/main/java/com/rusticisoftware/tincan/RemoteLRS.java

```diff
@@ -223,19 +220,113 @@ public void onHeaders(Response webResp) {
                 }
             }
 
-            if (req.getContentType() != null) {
-                webReq.header("Content-Type", req.getContentType());
+            OutputStreamContentProvider content = new OutputStreamContentProvider();
+            InputStreamResponseListener listener = new InputStreamResponseListener();
+
+            try (OutputStream output = content.getOutputStream()) {
+                if (req.getPartList() == null || req.getPartList().size() <= 0) {
+                    if (req.getContentType() != null) {
+                        webReq.header("Content-Type", req.getContentType());
+                    }
+                    else if (req.getContentType() != "GET") {
+                        webReq.header("Content-Type", "application/octet-stream");
+                    }
+
+                    webReq.content(content).send(listener);
+
+                    if (req.getContent() != null) {
+                        output.write(req.getContent());
+                    }
+
+                    output.close();
+                }
+                else {
+
+                    MultiPartOutputStream multiout = new MultiPartOutputStream(output);
+
+                    webReq.header("Content-Type", "multipart/mixed; boundary=" + multiout.getBoundary());
+                    webReq.content(content).send(listener);
+
+                    if (req.getContentType() != null) {
+                        multiout.startPart(req.getContentType());
+                    } else {
+                        multiout.startPart("application/octet-stream");
+                    }
+
+                    if (req.getContent() != null) {
+                        multiout.write(req.getContent());
+                    }
+
+                    for (HTTPPart part : req.getPartList()) {
+                        multiout.startPart(part.getContentType(), new String[]{
+                                "Content-Transfer-Encoding: binary",
+                                "X-Experience-API-Hash: " + part.getSha2()});
+                        multiout.write(part.getContent());
+                    }
+                    multiout.close();
+                }
             }
-            else {
-                webReq.header("Content-Type", "application/octet-stream");
+
+            Response httpResponse = listener.get(5, TimeUnit.SECONDS);
```

Comment:
@brianjmiller Alright, after reexamination, this may be the only way to go. There is a method, `listener.await()`, that will wait either until the entire response (headers and content) has arrived OR the timeout happens. I think making the timeout limit configurable would be the best option.